frontends/ast/simplify.cc

```diff
@@ -1292,6 +1292,12 @@ bool AstNode::simplify(bool const_fold, int stage, int width_hint, bool sign_hin
 				else if (contains_unbased_unsized(value))
 					// unbased unsized literals extend to width of the context
 					lookup_suggested = true;
+				else if (value->type == AST_TO_UNSIGNED)
+					// inner expression may be signed by default
+					lookup_suggested = true;
+				else if (value->type == AST_CONCAT && value->children.size() == 1)
+					// concat of a single expression is equivalent to $unsigned
+					lookup_suggested = true;
```

Comment:
OK, so suggesting a lookup here will make us indirect all port connections that are not connections to a plain identifier.

Reply:
Correct. Is this problematic? Should I clarify the surrounding code?

Reply:
Well, the intent wasn't really obvious the first moment I looked at this `if (type == AST_CELL) {` branch, so any clarification would be nice. Though I wouldn't consider it required for the PR.